woocommerce/woocommerce-ios
# [Custom amounts M4] Update custom amount card UI and deletion flow

https://github.com/woocommerce/woocommerce-ios/pull/11239 needs to be merged first 

## Description
This PR updates the UI design for: 
- The custom amount card row that can be found in the editable order: We simplify it by only displaying the Title, Amount, and (pencil) Edit button.
- The editable custom amount view: We add a delete button that renders only when editing an existing custom amount.

| Order editing view | Custom amount editing view |
|--------|--------|
| ![simulator_screenshot_99094B11-88CB-492D-AE05-535A578755F0](https://github.com/woocommerce/woocommerce-ios/assets/3812076/a742cc7b-e130-4d8d-838e-85a88d966420) | ![simulator_screenshot_01CB50D3-1025-4804-836E-A89E204BE9D8](https://github.com/woocommerce/woocommerce-ios/assets/3812076/088830f3-ff41-4efd-9b86-110f72781bc6) |

## Testing instructions
1. Add a custom amount to an Order
2. Observe how there's no "Delete Custom Amount" button
3. Fill up the custom amount and save it
4. Confirm that the design matches and multiline custom amount titles doesn't break the presentation in the Editable Order UI
5. Feel free to add multiple custom amounts
6. Tap on the edit button (pencil) on one of them
7. Observe how a "Delete Custom Amount" button is rendered
8. Observe how tapping on the deletion button dismisses the view and removes the custom amount from the editable order, as well as tracks the deletion events:
```
🔵 Tracked order_creation_remove_custom_amount_tapped, properties: [AnyHashable("blog_id"): 215063064, AnyHashable("site_url"): "https://indiemelon.mystagingwebsite.com", AnyHashable("is_wpcom_store"): false, AnyHashable("was_ecommerce_trial"): false, AnyHashable("plan"): "jetpack_security_daily"]

🔵 Tracked order_fee_remove, properties: [AnyHashable("is_wpcom_store"): false, AnyHashable("plan"): "jetpack_security_daily", AnyHashable("blog_id"): 215063064, AnyHashable("was_ecommerce_trial"): false, AnyHashable("flow"): "creation", AnyHashable("site_url"): "https://indiemelon.mystagingwebsite.com"]
```


![Simulator Screen Recording - iPhone 15 - 2023-11-23 at 12 52 43](https://github.com/woocommerce/woocommerce-ios/assets/3812076/e1e4f47b-9a3c-410b-a1eb-ee44be6de263)
